pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -158,11 +163,45 @@ func isRequiredConfigPresent(config []byte, isNotSingleNode bool) error {
 			if !ok {
 				return fmt.Errorf("%v is not a slice", strings.Join(requiredPath, "."))
 			}
-			if len(configValSlice) < 3 {
-				return fmt.Errorf("%v has less than three endpoints: %v", strings.Join(requiredPath, "."), configValSlice)
+
+			if err := etcdEndpointsPresent(c.configMapLister, configValSlice, requiredPath); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// etcdEndpointsPresent compares the list of endpoints exposed by
+// cluster-etcd-operator with the list of URLs in the etcd-servers config. If
+// less than two etcd live endpoints (not localhost and bootstrap) are present
+// in the config, it returns an error because the configuration does not
+// statisfy HA requirements.
+func etcdEndpointsPresent(configMapLister corev1listers.ConfigMapLister, config []interface{}, configPath []string) error {
+	etcdEndpointsCM, err := configMapLister.ConfigMaps(etcdEndpointNamespace).Get(etcdEndpointName)
```

Comment:
Is the reason for checking the configmap here to exclude loopback endpoints? It's our config observer that might be adding the loopback endpoint anyway. I think I would prefer parsing each URL from the arguments and comparing the hosts vs. introducing a new API dependency to this function.